app/controllers/jobseekers/job_applications_controller.rb

```diff
@@ -80,6 +84,22 @@ def withdraw
 
   private
 
+  def completed_steps_valid?
+    # Check that all completed steps are valid, in case we have changed the validations since the step was completed.
+    # NB: Only validates top-level step forms. Does not validate individual qualifications, employments, or references.
+    job_application.completed_steps.all? do |step|
+      step_valid?("jobseekers/job_application/#{step}_form".camelize.constantize)
+    end
+  end
+
+  def step_valid?(step_form)
+    form = step_form.new(job_application.slice(*send("#{step_form.to_s.underscore.split('/').last.split('_form').first}_fields")))
```

Comment:
Maybe if `completed_steps_valid?` passes in `step` to this method, and then this method constructs the class, this would be simpler?

I know we've got this multi-step form validation pattern elsewhere, but I'd love for us to have a stab at revisiting it one day (not this PR 😄 )